app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterFragment.kt

```diff
@@ -109,28 +110,39 @@ open class ChapterFragment : Fragment(), AudioListener {
 
             val wordViewAdapter =
                 WordViewAdapter(readingLevelPosition, object : WordViewAdapter.OnItemClickListener {
-                    override fun onItemClick(wordGson: WordGson?, view: View?, position: Int) {
-                        wordGson ?: return
-                        Log.i(TAG, "onClick")
-                        Log.i(TAG, "wordGson.text: ${wordGson.text}")
-
-                        activity?.supportFragmentManager?.let { fragmentManager ->
-                            WordDialogFragment.newInstance(wordGson.id)
-                                .show(fragmentManager, "dialog")
+                    override fun onItemClick(
+                        wordGson: WordGson?,
+                        text: String,
+                        view: View?,
+                        position: Int
+                    ) {
+                        Log.i(TAG, "onClick wordGson: $wordGson. text: $text")
+                        var spokenText = text
+                        var utteranceId = UUID.randomUUID().toString()
```

Comment:
Since we are already using `utteranceId = "word_" + wordGson.id` below without a random ID, it would be more consistent if we do something similar for words without a `WordGson` representation:

```suggestion
                        var utteranceId = "word_" + wordText
```

Reply:
@jo-elimu I think the usage of `word_` + `wordGson.id`  is already not a good example for us to follow, because by definition: `utteranceId` is `An unique identifier for this request.` (the text-to-speech request).
And adding such `word_` prefix doesn't make (or contribute at any extent to) the uniqueness of the request.
Therefore, I believe adding the prefix to the `utteranceId` here makes the code longer without creating any meaning, even consistency.